README
Net-Cassandra-libcassandra version 0.01
=======================================

The README is used to introduce the module and provide instructions on
how to install the module, any machine dependencies it may have (for
example C compilers and installed libraries) and any other information
that should be provided before the module is installed.

A README file is required for CPAN modules since CPAN extracts the
README file from a module distribution so that people browsing the
archive can use it get an idea of the modules uses. It is usually a
good idea to provide version information here so that people can
decide whether fixes for the module are worth downloading.

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

DEPENDENCIES

This module requires these other modules and libraries:

  blah blah blah

COPYRIGHT AND LICENCE

Put the correct copyright and licence information here.

Copyright (C) 2010 by A. U. Thor

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.8 or,
at your option, any later version of Perl 5 you may have available.




Changes
Revision history for Perl extension Net::Cassandra::libcassandra.

0.01  Tue Jun  1 15:01:56 2010
	- original version; created by h2xs 1.23 with options
		-A -n Net::Cassandra::libcassandra



Makefile.PL
use 5.008008;
use ExtUtils::MakeMaker;
# See lib/ExtUtils/MakeMaker.pm for details of how to influence
# the contents of the Makefile that is written.
WriteMakefile(
    NAME              => 'Net::Cassandra::libcassandra',
    VERSION_FROM      => 'lib/Net/Cassandra/libcassandra.pm', # finds $VERSION
    PREREQ_PM         => {}, # e.g., Module::Name => 1.1
    ($] >= 5.005 ?     ## Add these new keywords supported since 5.005
      (ABSTRACT_FROM  => 'lib/Net/Cassandra/libcassandra.pm', # retrieve abstract from module
       AUTHOR         => 'A. U. Thor <stanaka@osuosl.org>') : ()),
    #LIBS              => ['-lthrift -lcassandra'], # e.g., '-lm'
    LIBS              => ['-lcassandra'], # e.g., '-lm'
    #LIBS              => ['-lthrift'], # e.g., '-lm'
    DEFINE            => '', # e.g., '-DHAVE_SOMETHING'
    INC               => '-I. -I/usr/include/thrift', # e.g., '-I. -I/usr/include/other'
	# Un-comment this if you add C files to link with later:
    # OBJECT            => '$(O_FILES)', # link all the C files too
    CC => '/usr/bin/g++',
    LD => '/usr/bin/g++',
    XSOPT => '-C++',
    TYPEMAPS => ['perlobject.map'],
);



examples/simple.pl
use lib qw(./lib ./blib/arch/auto/Net/Cassandra/libcassandra);
use Net::Cassandra::libcassandra;
use Data::Dumper;

my $cassandra = Net::Cassandra::libcassandra::new('localhost', 9160);

warn $cassandra->getClusterName();
warn Data::Dumper::Dumper($cassandra->getKeyspaces());
warn Data::Dumper::Dumper($cassandra->getTokenMap(0));

my $keyspace = $cassandra->getKeyspace("Keyspace1");
warn $keyspace->getName();
$keyspace->insertColumn("key", "Standard1", "", "name", "value");

my $res = $keyspace->getColumnValue("key", "Standard1", "", "name");
warn "Value in column retrieved is: ". $res;

$keyspace->remove("key", "Standard1", "", "name");
warn "deleted";

eval {
    $res = $keyspace->getColumnValue("key", "Standard1", "name");
    warn "Value in column retrieved is: ". $res;
};
warn $@ if $@;




lib/Net/Cassandra/libcassandra/Keyspace.pm
package Net::Cassandra::libcassandra::Keyspace;

use 5.008008;
use strict;
use warnings;
use Net::Cassandra::libcassandra;

sub insertColumn {
    my ($self, $key, $cf, $scn, $cn, $value) = @_;
    unless($value){
        Net::Cassandra::libcassandra::keyspace_insertColumn2($self, $key, $cf, $scn, $cn);
    } else {
        Net::Cassandra::libcassandra::keyspace_insertColumn($self, $key, $cf, $scn, $cn, $value);
    }
}

sub remove {
    my ($self, $key, $cf, $scn, $cn) = @_;
    unless($cn){
        Net::Cassandra::libcassandra::keyspace_remove($self, $key, $cf);
    } else {
        Net::Cassandra::libcassandra::keyspace_remove2($self, $key, $cf, $scn, $cn);
    }
}

sub removeColumn {
    my ($self, $key, $cf, $scn, $cn) = @_;
    Net::Cassandra::libcassandra::keyspace_removeColumn($self, $key, $cf, $scn, $cn);
}

sub removeSuperColumn {
    my ($self, $key, $cf, $scn) = @_;
    Net::Cassandra::libcassandra::keyspace_removeSuperColumn($self, $key, $cf, $scn);
}

sub getColumn {
    my ($self, $key, $cf, $scn, $cn) = @_;
    unless($cn){
        Net::Cassandra::libcassandra::keyspace_getColumn2($self, $key, $cf, $scn);
    } else {
        Net::Cassandra::libcassandra::keyspace_getColumn($self, $key, $cf, $scn, $cn);
    }
}

sub getColumnValue {
    my ($self, $key, $cf, $scn, $cn) = @_;
    unless($cn){
        Net::Cassandra::libcassandra::keyspace_getColumnValue2($self, $key, $cf, $scn);
    } else {
        Net::Cassandra::libcassandra::keyspace_getColumnValue($self, $key, $cf, $scn, $cn);
    }
}

sub getSuperColumn {
    my ($self, $key, $cf, $scn) = @_;
    Net::Cassandra::libcassandra::keyspace_getSuperColumn($self, $key, $cf, $scn);
}

sub getSliceNames {
    my ($self, $key, $col_parent, $pred) = @_;
    Net::Cassandra::libcassandra::keyspace_getSliceNames($self, $key, $col_parent, $pred);
}

sub getSliceRange {
    my ($self, $key, $col_parent, $pred) = @_;
    Net::Cassandra::libcassandra::keyspace_getSliceRange($self, $key, $col_parent, $pred);
}

sub getCount {
    my ($self, $key, $col_parent) = @_;
    Net::Cassandra::libcassandra::keyspace_getCount($self, $key, $col_parent);
}

sub getName {
    my ($self) = @_;
    Net::Cassandra::libcassandra::keyspace_getName($self);
}

1;


lib/Net/Cassandra/libcassandra.pm
package Net::Cassandra::libcassandra;

use 5.008008;
use strict;
use warnings;
use Net::Cassandra::libcassandra::Keyspace;

require Exporter;

our @ISA = qw(Exporter);

# Items to export into callers namespace by default. Note: do not export
# names by default without a very good reason. Use EXPORT_OK instead.
# Do not simply export all your public functions/methods/constants.

# This allows declaration	use Net::Cassandra::libcassandra ':all';
# If you do not need this, moving things directly into @EXPORT or @EXPORT_OK
# will save memory.
our %EXPORT_TAGS = ( 'all' => [ qw(
	
) ] );

our @EXPORT_OK = ( @{ $EXPORT_TAGS{'all'} } );

our @EXPORT = qw(
	
);

our $VERSION = '0.01';

require XSLoader;
XSLoader::load('Net::Cassandra::libcassandra', $VERSION);

# Preloaded methods go here.

1;
__END__
# Below is stub documentation for your module. You'd better edit it!

=head1 NAME

Net::Cassandra::libcassandra - Perl binding for Cassandra with libcassandra.

=head1 SYNOPSIS

  use Net::Cassandra::libcassandra;

  my $cassandra = Net::Cassandra::libcassandra::new('localhost', 9160);

  warn $cassandra->getClusterName();
  warn $cassandra->getKeyspaces();
  warn $cassandra->getTokenMap(0);

  my $keyspace = $cassandra->getKeyspace("Keyspace1");
  warn $keyspace->getName();
  $keyspace->insertColumn("aa", "Standard1", "cc", "dd");

  my $res = $keyspace->getColumnValue("aa", "Standard1", "cc");
  warn "Value in column retrieved is: ". $res;

=head1 DESCRIPTION

This module provides a Perl interface to Cassandra via libcassandra.
It uses XS for communicating with libcassandra.

=head1 SEE ALSO

Net::Cassandra, Net::Cassandra::Easy

=head1 AUTHOR

Shinji Tanaka <shinji.tanaka@gmail.com>

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2010 by Shinji Tanaka

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.8 or,
at your option, any later version of Perl 5 you may have available.

=cut
